### PR TITLE
CHI-2974 CLI tool for republishing contacts to s3 lake 

### PR DESCRIPTION
**Pull request by mythilytm**
## Description

- This PR adds the cli tool for a new admin endpoint `/contacts/republish`. 
- See documentation on how to use https://techmatters.app.box.com/notes/1722659624579

### Checklist
- [ ] Corresponding issue has been opened
- [ ] New tests added
- [ ] Feature flags / configuration added

### Other Related Issues

None

### Verification steps
- Ensure this branch is deployed to the appropriate environment
- Log into devops instance to run the 'republish'. 
  - `npm run admin-cli republish hrm -- --contacts -e <env> -r <region> -a <accountSid> -f <dateFrom>-t <dateTo>`
  - See documentation on how to use https://techmatters.app.box.com/notes/1722659624579
- To Verify, check the data lake s3 bucket to ensure the data has been updated
### AFTER YOU MERGE

1. Cut a release tag using the GitHub workflow. Wait for it to complete and notify in the #aselo-deploys Slack channel.
2. Comment on the ticket with the release tag version AND any additional instructions required to configure an environment to test the changes.
3. Only then move the ticket into the QA column in JIRA

You are responsible for ensuring the above steps are completed. If you move a ticket into QA without advising what version to test, the QA team will assume the latest tag has the changes. If it does not, the following confusion is on you! :-P

### CODE REVIEW

File: hrm-domain/hrm-core/notifications/entityChangeNotify.ts [hunk 1]
```diff
@@ -127,6 +127,16 @@ const publishEntityToSearchIndex = async (
       payload: { accountSid, id: entity.id.toString(), operation },
       messageGroupId,
     });
+  } else if (operation === 'republish' && entityType === 'contact') {
```
Comment (GPaoloni) [inline hrm-domain/hrm-core/notifications/entityChangeNotify.ts:130]: I think is safe to just check for `operation` here.

**Review by GPaoloni — COMMENTED** [review]
I DMed you with the possible solution to the build errors. One small comment otherwise.

File: .github/workflows/generate-ai-training-ecs.yml [hunk 2]
```diff
@@ -55,7 +55,7 @@ jobs:
           SUBNETS=$(if [ ${{ inputs.environment}} == "development" ]; then echo "subnet-034e5c652dbad09dd"; else echo "subnet-00b4e88c0ea178f3d"; fi)
           SECURITY_GROUPS=($(if [ ${{ inputs.environment}} == "development" ]; then echo "sg-09194f9a648baf082 sg-047498d4c7b2cedd8 sg-06d6458accc0ec5ed sg-0ace4338c75e5d3de"; 
             else echo "sg-02fd053fe7b4660ed sg-0e7119c4423f2c7b0 sg-066d0055caa7cbe90 sg-0c7fb74e11cb64ff8"; fi))
-          COMMAND="npm run start:generate-ai-training-set ${{ inputs.environment }} ${{ inputs.helpline-shortcodes }} tl-aselo-ai-${{ inputs.environment}}"
+          COMMAND="npm run start:generate-ai-training-set ${{ inputs.environment }} ${{ inputs.helpline-shortcodes }} tl-aselo-ai-${{ inputs.environment }}$(if [ ${{ inputs.environment }} == "production" ]; then echo "-us-east-1"; fi)"
```
Comment (GPaoloni) [inline .github/workflows/generate-ai-training-ecs.yml:58]: Is this change intentional in this branch?

File: hrm-domain/hrm-core/contact/contactDataAccess.ts [hunk 3]
```diff
@@ -369,6 +369,7 @@ export const streamContactsForReindexing = ({
   viewPermissions: TKConditionsSets<'contact'>;
   batchSize?: number;
 }): Promise<NodeJS.ReadableStream> => {
+  console.log('>>> streamContactsAfterNotified', searchParameters, viewPermissions, user);
```
Comment (GPaoloni) [inline hrm-domain/hrm-core/contact/contactDataAccess.ts:372]: :broom:

**Review by GPaoloni — APPROVED** [review]
LGTM, just two small things:
- I pushed a change changing a bit some types. Sorry that I did this, it was just easier than to ask you changing the code, but feel free to revert if that is not correct.
- ~Can you please update the scripts README with the instructions how to use `republish`?~ I said nothing, you did it already :yum: Thanks!

File: .github/workflows/generate-ai-training-ecs.yml (hunk 2 above)
Comment (mythilytm) [reply .github/workflows/generate-ai-training-ecs.yml:58]: It was part of a different PR that got added when I merged master.

**Comment by mythilytm** [comment]
> * I pushed a change changing a bit some types. Sorry that I did this, it was just easier than to ask you changing the code, but feel free to revert if that is not correct.
Thanks
